include/flatbuffers/flexbuffers.h

```diff
@@ -602,7 +607,7 @@ class Reference {
 
   template<typename T> bool Mutate(const uint8_t *dest, T t, size_t byte_width,
                                    BitWidth value_width) {
-    auto fits = static_cast<size_t>(1U << value_width) <= byte_width;
+    auto fits = static_cast<size_t>(static_cast<size_t>(1U) << value_width) <= byte_width;
```

Comment:
this seems a bit crazy having 2 nested casts.. if you just remove the `U` from the original code, does that work?